store/store_test.go

```diff
@@ -2582,6 +2582,54 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryListRefreshRetryOnEOF(c *
 	c.Assert(results[0].Name(), Equals, "hello-world")
 }
 
+func (t *remoteRepoTestSuite) TestUbuntuStoreUnexpectedEOFhandling(c *C) {
+	mockServerHelper := func(failures int) error {
+		var mockServer *httptest.Server
+		n := 0
+		mockServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
maybe just `:=` so that we don't need the `var mockServer *httptest.Server` line above?